### PR TITLE
Remove the `proxy.disableIdentity` config

### LINKED ISSUE #7724: Remove `identityDisabled` configurations
The proxy no longer supports operating without identity. We should remove helm templating conditionals that allow identity to be disabled so that it's not possible to produce a broken configuration.

### PR DESCRIPTION
**Pull request by alpeb**
Fixes #7724

Also:
- Removed the `linkerd.io/identity-mode` annotation.
- Removed the `config.linkerd.io/disable-identity` annotation.
- Removed the `linkerd.proxy.validation` template partial, which only
  made sense when `proxy.disableIdentity` was `true`.
- TestInjectManualParams now requires to hit the cluster to retrieve the
  trust root.

### CODE REVIEW

File: cli/cmd/options.go [hunk 1]
```diff
@@ -469,12 +469,6 @@ func makeInjectFlags(defaults *l5dcharts.Values) ([]flag.Flag, *pflag.FlagSet) {
 				return nil
 			}),
 
-		flag.NewBoolFlag(injectFlags, "disable-identity", defaults.Proxy.DisableIdentity,
-			"Disables resources from participating in TLS identity", func(values *l5dcharts.Values, value bool) error {
-				values.Proxy.DisableIdentity = value
-				return nil
-			}),
```
Comment (olix0r) [inline cli/cmd/options.go:476]: Should we retain this flag and print an error?

Comment (alpeb) [reply cli/cmd/options.go:476]: That would makes sense to me if we'd be removing some functionality, but this has been broken for a long time, so why not just get rid of it?

Comment (olix0r) [reply cli/cmd/options.go:476]: This changed in 2.11, so it wasn't that long ago.. we just missed this as a part of the 2.11 release. I don't feel strongly, though.

**Review by adleong — APPROVED** [review]

**Review by olix0r — APPROVED** [review]

File: cli/cmd/options.go (hunk 1 above)
Comment (alpeb) [reply cli/cmd/options.go:476]: Ok, I'll add it back, erroring with a message.

**Pull request merged**